pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -104,70 +204,51 @@ func (pl *VolumeZone) Filter(ctx context.Context, _ *framework.CycleState, pod *
 		return nil
 	}
 
-	for i := range pod.Spec.Volumes {
-		volume := pod.Spec.Volumes[i]
-		if volume.PersistentVolumeClaim == nil {
-			continue
-		}
-		pvcName := volume.PersistentVolumeClaim.ClaimName
-		if pvcName == "" {
-			return framework.NewStatus(framework.UnschedulableAndUnresolvable, "PersistentVolumeClaim had no name")
-		}
-		pvc, err := pl.pvcLister.PersistentVolumeClaims(pod.Namespace).Get(pvcName)
-		if s := getErrorAsStatus(err); !s.IsSuccess() {
-			return s
-		}
-
-		pvName := pvc.Spec.VolumeName
-		if pvName == "" {
-			scName := storagehelpers.GetPersistentVolumeClaimClass(pvc)
-			if len(scName) == 0 {
-				return framework.NewStatus(framework.UnschedulableAndUnresolvable, "PersistentVolumeClaim had no pv name and storageClass name")
-			}
-
-			class, err := pl.scLister.Get(scName)
-			if s := getErrorAsStatus(err); !s.IsSuccess() {
-				return s
-			}
-			if class.VolumeBindingMode == nil {
-				return framework.NewStatus(framework.UnschedulableAndUnresolvable, fmt.Sprintf("VolumeBindingMode not set for StorageClass %q", scName))
-			}
-			if *class.VolumeBindingMode == storage.VolumeBindingWaitForFirstConsumer {
-				// Skip unbound volumes
-				continue
-			}
+	var podPVTopologies []pvTopology
 
-			return framework.NewStatus(framework.UnschedulableAndUnresolvable, "PersistentVolume had no name")
+	state, err := getStateData(cs)
+	if err != nil {
+		//Fallback to calculate pv list here
+		var status *framework.Status
+		podPVTopologies, status = pl.getPVbyPod(ctx, pod)
+		if !status.IsSuccess() {
+			return status
 		}
+	} else {
+		podPVTopologies = state.podPVTopologies
+	}
 
-		pv, err := pl.pvLister.Get(pvName)
-		if s := getErrorAsStatus(err); !s.IsSuccess() {
-			return s
+	for _, pvTopology := range podPVTopologies {
+		nodeV := nodeConstraints[pvTopology.labelKey]
+		volumeVSet, err := volumehelpers.LabelZonesToSet(pvTopology.labelValue)
```

Comment:
this should be done in Prefilter

Reply:
Save `volumeVSet` in stateDate before filter?